src/core/types.ts

```diff
@@ -3,7 +3,7 @@ import type { QueryBehavior } from './query'
 import type { RetryValue, RetryDelayValue } from './retryer'
 import type { QueryFilters } from './utils'
 
-export type QueryKey = string | unknown[]
+export type QueryKey = string | ReadonlyArray<unknown>
```

Comment:
minor nitpick since we use `[]` everywhere else would be to change it to `readonly unknown[]`

Reply:
done